[Linux][GTK] Build error on Ubuntu 18.04

## Steps to Reproduce

You probably already know, it happens build error on Ubuntu 18.04.
Of course, by removing `-Werror` option of CMake, the build was successful.

Do you get build error?

1.  $ git clone https://github.com/google/flutter-desktop-embedding.git
2.  $ cd flutter-desktop-embedding/testbed
3.  $ mv linux linux-glfw
4.  $ mv gtk linux
5.  $ flutter run -d linux --verbose

**Actual results:** <!-- what did you see? -->
```
/home/matsu/workspace/flutter-desktop-embedding/testbed/linux/main.cc
[   +5 ms] /home/matsu/workspace/flutter-desktop-embedding/testbed/linux/main.cc:7:1: error: unused function 'FL_IS_APPLICATION'
[-Werror,-Wunused-function]
[        ] G_DECLARE_FINAL_TYPE(FlApplication, fl_application, FL, APPLICATION,
[        ] ^
[        ] /usr/include/glib-2.0/gobject/gtype.h:1405:26: note: expanded from macro 'G_DECLARE_FINAL_TYPE'
[        ]   static inline gboolean MODULE##_IS_##OBJ_NAME (gpointer ptr) {                                         \
[   +5 ms]                          ^
[        ] <scratch space>:48:1: note: expanded from here
[        ] FL_IS_APPLICATION
[   +3 ms] ^
[        ] 1 error generated.
[  +35 ms] [4/14] Building CXX object CMakeFiles/testbed.dir/window_configuration.cc.o
[        ] ninja: build stopped: subcommand failed.
[  +12 ms] Building Linux application... (completed in 12.5s)
[        ] Exception: Build process failed
[   +3 ms] "flutter run" took 13,375ms.
[  +16 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:578:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:829:18)
           #3      _rootRunUnary (dart:async/zone.dart:1198:47)
           #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
           #12     _rootRunUnary (dart:async/zone.dart:1198:47)
           #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #18     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #19     _rootRun (dart:async/zone.dart:1190:13)
           #20     _CustomZone.run (dart:async/zone.dart:1093:19)
           #21     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #22     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #23     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #25     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #26     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
           
           
[ +207 ms] ensureAnalyticsSent: 199ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[  +11 ms] Shutdown hooks complete
[        ] exiting with code 1
```

<details>
  <summary>flutter doctor log</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 1.19.0-6.0.pre.42, on Linux, locale ja_JP.UTF-8)
 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    ✗ Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[!] Android Studio (version 3.5)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.45.1)
[✓] Connected device (3 available)

! Doctor found issues in 2 categories.
```
</details>

Thank you.

--- a/testbed/gtk/CMakeLists.txt
+++ b/testbed/gtk/CMakeLists.txt
@@ -33,6 +33,7 @@ pkg_check_modules(GTK REQUIRED IMPORTED_TARGET gtk+-3.0)
 # Application build
 add_executable(${BINARY_NAME}
   "main.cc"
+  "fl_application.cc"
   "window_configuration.cc"
   "${FLUTTER_MANAGED_DIR}/gtk_plugin_registrant.cc"
 )

--- a/testbed/gtk/fl_application.cc
+++ b/testbed/gtk/fl_application.cc
@@ -1,0 +1,42 @@
+#include "fl_application.h"
+
+#include <flutter_linux/flutter_linux.h>
+
+#include "flutter/gtk_plugin_registrant.h"
+#include "window_configuration.h"
+
+struct _FlApplication {
+  GtkApplication parent_instance;
+};
+
+G_DEFINE_TYPE(FlApplication, fl_application, GTK_TYPE_APPLICATION)
+
+// Implements GApplication::activate.
+static void fl_application_activate(GApplication* application) {
+  GtkWindow* window =
+      GTK_WINDOW(gtk_application_window_new(GTK_APPLICATION(application)));
+  gtk_widget_show(GTK_WIDGET(window));
+  gtk_widget_set_size_request(GTK_WIDGET(window), kFlutterWindowWidth,
+                              kFlutterWindowHeight);
+  gtk_window_set_title(window, kFlutterWindowTitle);
+
+  g_autoptr(FlDartProject) project = fl_dart_project_new();
+
+  FlView* view = fl_view_new(project);
+  gtk_widget_show(GTK_WIDGET(view));
+  gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));
+
+  fl_register_plugins(FL_PLUGIN_REGISTRY(view));
+
+  gtk_widget_grab_focus(GTK_WIDGET(view));
+}
+
+static void fl_application_class_init(FlApplicationClass* klass) {
+  G_APPLICATION_CLASS(klass)->activate = fl_application_activate;
+}
+
+static void fl_application_init(FlApplication* self) {}
+
+FlApplication* fl_application_new() {
+  return FL_APPLICATION(g_object_new(fl_application_get_type(), nullptr));
+}

--- a/testbed/gtk/fl_application.h
+++ b/testbed/gtk/fl_application.h
@@ -1,6 +1,18 @@
+#ifndef FLUTTER_FL_APPLICATION_H_
+#define FLUTTER_FL_APPLICATION_H_
+
 #include <gtk/gtk.h>
 
 G_DECLARE_FINAL_TYPE(FlApplication, fl_application, FL, APPLICATION,
                      GtkApplication)
 
+/**
+ * fl_application_new:
+ *
+ * Creates a new Flutter application.
+ *
+ * Returns: a new #FlApplication.
+ */
 FlApplication* fl_application_new();
+
+#endif  // FLUTTER_FL_APPLICATION_H_

--- a/testbed/gtk/fl_application.h
+++ b/testbed/gtk/fl_application.h
@@ -1,0 +1,6 @@
+#include <gtk/gtk.h>
+
+G_DECLARE_FINAL_TYPE(FlApplication, fl_application, FL, APPLICATION,
+                     GtkApplication)
+
+FlApplication* fl_application_new();

--- a/testbed/gtk/main.cc
+++ b/testbed/gtk/main.cc
@@ -1,47 +1,4 @@
-#include <flutter_linux/flutter_linux.h>
-#include <gtk/gtk.h>
-
-#include "flutter/gtk_plugin_registrant.h"
-#include "window_configuration.h"
-
-G_DECLARE_FINAL_TYPE(FlApplication, fl_application, FL, APPLICATION,
-                     GtkApplication)
-
-struct _FlApplication {
-  GtkApplication parent_instance;
-};
-
-G_DEFINE_TYPE(FlApplication, fl_application, GTK_TYPE_APPLICATION)
-
-// Implements GApplication::activate.
-static void fl_application_activate(GApplication* application) {
-  GtkWindow* window =
-      GTK_WINDOW(gtk_application_window_new(GTK_APPLICATION(application)));
-  gtk_widget_show(GTK_WIDGET(window));
-  gtk_widget_set_size_request(GTK_WIDGET(window), kFlutterWindowWidth,
-                              kFlutterWindowHeight);
-  gtk_window_set_title(window, kFlutterWindowTitle);
-
-  g_autoptr(FlDartProject) project = fl_dart_project_new();
-
-  FlView* view = fl_view_new(project);
-  gtk_widget_show(GTK_WIDGET(view));
-  gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));
-
-  fl_register_plugins(FL_PLUGIN_REGISTRY(view));
-
-  gtk_widget_grab_focus(GTK_WIDGET(view));
-}
-
-static void fl_application_class_init(FlApplicationClass* klass) {
-  G_APPLICATION_CLASS(klass)->activate = fl_application_activate;
-}
-
-static void fl_application_init(FlApplication* self) {}
-
-static FlApplication* fl_application_new() {
-  return FL_APPLICATION(g_object_new(fl_application_get_type(), nullptr));
-}
+#include "fl_application.h"
 
 int main(int argc, char** argv) {
   g_autoptr(FlApplication) app = fl_application_new();

